### PR TITLE
optimize: optimize client batch sending.

### LINKED ISSUE #4156: 关于客户端批量发送的优化处理
当前RM客户端和TM客户端，默认都使用批量发送请求，其中TM客户端的请求消息会相对耗时，这会影响其他消息的响应时间。虽然RM客户端和TM客户端都有各自的批量发送线程模型，但只有一个批量发送的开关，无法精准控制RM客户端和TM客户端的批量发送请求。

### PR DESCRIPTION
**Pull request by objcoding**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?

fixes #4156

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java [hunk 1]
```diff
@@ -267,6 +270,24 @@ protected String getTransactionServiceGroup() {
         return transactionServiceGroup;
     }
 
+    @Override
+    public boolean isEnableClientBatchSendRequest() {
+        boolean isEnable = DefaultValues.DEFAULT_ENABLE_RM_CLIENT_BATCH_SEND_REQUEST;
+        // New configuration takes precedence
+        String rmConfig = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_RM_CLIENT_BATCH_SEND_REQUEST);
+        if (StringUtils.isNotBlank(rmConfig)) {
+            isEnable = Boolean.parseBoolean(rmConfig);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java:279]: return

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java [hunk 2]
```diff
@@ -267,6 +270,24 @@ protected String getTransactionServiceGroup() {
...
+        }
+        // Compatible with old configuration
+        // If the old configuration exists, use the old configuration
+        String config = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_RM_CLIENT_BATCH_SEND_REQUEST);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java:283]: ENABLE_CLIENT_BATCH_SEND_REQUEST

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 1 above)
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: 这里不为空是否直接return?

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 2 above)
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: ConfigurationKeys.ENABLE_CLIENT_BATCH_SEND_REQUEST

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 1 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 2 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java [hunk 3]
```diff
@@ -267,6 +270,23 @@ protected String getTransactionServiceGroup() {
...
+    @Override
+    public boolean isEnableClientBatchSendRequest() {
+        // New configuration takes precedence
+        String rmConfig = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_RM_CLIENT_BATCH_SEND_REQUEST);
+        if (StringUtils.isNotBlank(rmConfig)) {
+            return Boolean.parseBoolean(rmConfig);
+        }
+        // Compatible with old configuration
+        // If the old configuration exists, use the old configuration
+        String config = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_CLIENT_BATCH_SEND_REQUEST);
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java:282]: ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_CLIENT_BATCH_SEND_REQUEST,DefaultValues.DEFAULT_ENABLE_RM_CLIENT_BATCH_SEND_REQUES);

**Review by slievrly — COMMENTED** [review]
+  additional-spring-configuration-metadata.json
+ change log

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 1 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 3 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 2 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java [hunk 4]
```diff
@@ -267,6 +270,23 @@ protected String getTransactionServiceGroup() {
...
+    @Override
+    public boolean isEnableClientBatchSendRequest() {
+        // New configuration takes precedence
+        String newConfig = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_RM_CLIENT_BATCH_SEND_REQUEST);
+        if (StringUtils.isNotBlank(newConfig)) {
+            return Boolean.parseBoolean(newConfig);
+        }
+        // Compatible with old configuration
+        // If the old configuration exists, use the old configuration
+        String oldConfig = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_CLIENT_BATCH_SEND_REQUEST);
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java:282]: return ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.ENABLE_CLIENT_BATCH_SEND_REQUEST,DefaultValues.DEFAULT_ENABLE_RM_CLIENT_BATCH_SEND_REQUEST)

**Comment by slievrly** [comment]
image

File: core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java (hunk 4 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/RmNettyRemotingClient.java]: done.

**Comment by objcoding** [comment]
> image

done.

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**